frontend/src/libhoclient/host_orchestrator_client.go

```diff
@@ -504,13 +505,23 @@ func (c *HostOrchestratorServiceImpl) UploadFileWithOptions(uploadDir string, fi
 	if uploadOpts.BackOffOpts.MaxElapsedTime == 0 {
 		panic("MaxElapsedTime value cannot be zero")
 	}
-	uploader := &FilesUploader{
-		HTTPHelper:    c.HTTPHelper,
-		DumpOut:       c.HTTPHelper.Dumpster,
-		UploadDir:     uploadDir,
-		UploadOptions: uploadOpts,
+	lockRes := &hoapi.LockFileResponse{}
+	if err := c.HTTPHelper.NewPostRequest(fmt.Sprintf("/userartifacts/%s/%s/:lock", uploadDir, filepath.Base(filename)), nil).JSONResDo(lockRes); err != nil {
+		return err
+	}
+	if !lockRes.UploadCompleted {
```

Comment:
what happens if clients don't call the :lock endpoints and rather start uploading files directly? Are we expecting all clients to be good citizens? we cannot rely on that. Let's elaborate on what's the problem you're trying to solve first, hence we can discuss on the potential API changes.

Reply:
It's not about security, the comment is related to the API design itself. With these new methods, we cannot force every client implementation to call :lock before upload. We have control over the  HO behavior,  but every person can implement its own HO http client.